src/rascal/math/bessel.cc

```diff
@@ -129,6 +130,13 @@ void ModifiedSphericalBessel::calc(double distance, double fac_a) {
   this->bessel_arg = (2. * fac_a * distance) * this->x_v;
   this->bessel_arg_i = this->bessel_arg.inverse();
 
+  if (distance < SPHERICAL_BESSEL_FUNCTION_FTOL) {
+    this->bessel_values.setZero();
+    // 0th order approximation
+    this->bessel_values.col(0) = Eigen::exp(-this->x_v.square() * fac_a);
+    this->set_small_bessel_values_to_zero();
+  }
```

Comment:
I am not sure about the set_small_bessel_values_to_zero, why do you need it here ? Also this approximation implies that the other terms (order > 0) are 0 so it should stop here. The gradients would be 0 too.
```suggestion
    this->set_small_bessel_values_to_zero();
    if (this->compute_gradients) {
      this->bessel_gradients.setZero();
      // this->bessel_gradients.col(0) = - 2.*fac_a*distance * this->bessel_values.col(0);
    }
    return ;
  }
```